src/java/org/apache/cassandra/transport/Message.java

```diff
@@ -236,21 +242,16 @@ else if (StorageService.instance.shouldTraceProbablistically())
                 }
             }
 
-            Response response;
-            try
-            {
-                response = execute(queryState, queryStartNanoTime, shouldTrace);
-            }
-            finally
-            {
-                if (shouldTrace)
-                    Tracing.instance.stopSession();
-            }
-
-            if (isTraceable() && isTracingRequested())
-                response.setTracingId(tracingSessionId);
+            boolean finalShouldTrace = shouldTrace;
+            UUID finalTracingSessionId = tracingSessionId;
+            return executeAsync(queryState, queryStartNanoTime, shouldTrace)
+                   .whenComplete((result, ignored) -> {
```

Comment:
what about error handling in the async case?
previously, an exception would have been thrown and presumably something would happen.

now, nothing happens if the future completes exceptionally. Is that OK?

Reply:
This is the old code:
```
Response response;
try
{
    response = execute(queryState, queryStartNanoTime, shouldTrace);
}
finally
{
    if (shouldTrace)
        Tracing.instance.stopSession();
}

if (isTraceable() && isTracingRequested())
    response.setTracingId(tracingSessionId);

return response;
```

So there is no exception handling and any exception is eventually propagated.

The new async version works the same: the future eventually completes exceptionally and this is propagated upstream until it's handled in the `Dispatcher` (or in the persistence layer of Stargate).

Does that make sense?